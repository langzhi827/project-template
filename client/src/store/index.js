/**
 *  Author: harry.lang
 *  Date: 2017/11/24
 *  Description: Created by harrylang on 2017/11/24.
 */
import createStore from './createStore';

// Init store
const initialState = {};
const window = window || global;
const store = createStore(window.__INITIAL_STATE__ || initialState);

export default store;