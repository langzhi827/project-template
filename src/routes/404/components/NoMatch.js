/*
 * @Author: harry.lang 
 * @Date: 2018-03-20 14:54:22 
 * @Last Modified by: harry.lang
 * @Last Modified time: 2019-01-02 15:32:17
 */
import React, { PureComponent } from 'react';
import { DatePicker, Dropdown, Menu, Icon } from 'antd';
import CSSModules from 'react-css-modules';

import styles from '../assets/noMatch.css';
import img from '../assets/images/hehe.png';

class NoMatch extends PureComponent {

    componentDidMount() {
        const { getList } = this.props;
        getList();
    }

    changeTitle() {
        this.props.updateTitle('404-' + (+new Date));
    }

    render() {
        const { nomatch } = this.props;

        const menu = (
            <Menu>
                <Menu.Item>
                    <a target="_blank" rel="noopener noreferrer" href="http://www.alipay.com/">1st menu item</a>
                </Menu.Item>
                <Menu.Item>
                    <a target="_blank" rel="noopener noreferrer" href="http://www.taobao.com/">2nd menu item</a>
                </Menu.Item>
                <Menu.Item>
                    <a target="_blank" rel="noopener noreferrer" href="http://www.tmall.com/">3rd menu item</a>
                </Menu.Item>
            </Menu>
        );

        return <div styleName="no-match">
            <h2>{nomatch.title}</h2>

            <DatePicker />
            <Dropdown overlay={menu}>
                <a className="ant-dropdown-link" href="#">
                    Hover me <Icon type="down" />
                </a>
            </Dropdown>

            <img styleName="img" src={img} alt="404" onClick={this.changeTitle.bind(this)} />

            <h3>No match for <code>{this.props.location.pathname}</code></h3>
        </div>;
    }
}

export default CSSModules(NoMatch, styles);
