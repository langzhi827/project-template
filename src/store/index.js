import createStore from './createStore';

// Init store
const initialState = {};
const store = createStore(initialState);

export default store;