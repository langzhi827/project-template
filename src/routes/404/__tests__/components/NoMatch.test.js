import React from 'react';
import { render, mount, shallow } from 'enzyme';
import NoMatch from '../../components/NoMatch';

import { Provider } from 'react-redux';
import store, { initialState } from '../store';

describe('NoMatch', () => {
    const props = {
        location: { pathname: '/404' }
    };

    describe('renders correctly', () => {
        const wrapper = render(
            <Provider store={store}>
                <NoMatch {...props} />
            </Provider>
        );

        it('snapshot', () => {
            expect(wrapper).toMatchSnapshot();
        });

        it('default title', () => {
            expect(wrapper.find('h2').text()).toEqual(initialState.noMatch.title);
        });
    })

    describe('mount correctly', () => {
        const wrapper = mount(
            <Provider store={store} >
                <NoMatch  {...props} />
            </Provider>
        );

        it('throw exception', () => {
            expect(() => wrapper).not.toThrow();
        });

        it('click img', () => {
            wrapper.find('img').simulate('click');
            expect(wrapper.find('h2').text()).toEqual(store.getState().noMatch.title);
        });
    })
});