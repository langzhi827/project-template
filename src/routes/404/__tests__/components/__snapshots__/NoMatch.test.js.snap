// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoMatch renders correctly snapshot 1`] = `
<div
  class="noMatch"
>
  <h2>
    404
  </h2>
  <img
    alt="404"
    class="img-2q9nM"
    src="hehe.png"
  />
  <h3>
    No match for 
    <code>
      /404
    </code>
  </h3>
</div>
`;
